intellij-plugin-structure/structure-ide/src/main/java/com/jetbrains/plugin/structure/ide/ProductInfoBasedIdeManager.kt

```diff
@@ -155,16 +174,24 @@ class ProductInfoBasedIdeManager : IdeManager() {
     return IdeVersion.createIdeVersion(versionString)
   }
 
-  private fun getResourceResolver(layoutComponent: LayoutComponent, idePath: Path): NamedResourceResolver? {
-    return if (layoutComponent is LayoutComponent.Classpathable) {
-      val itemJarResolvers = layoutComponent.getClasspath().map { jarPath: Path ->
-        val fullyQualifiedJarFile = idePath.resolve(jarPath)
-        NamedResourceResolver(layoutComponent.name + "#" + jarPath, JarFilesResourceResolver(listOf(fullyQualifiedJarFile)))
-      }
-      NamedResourceResolver(layoutComponent.name, CompositeResourceResolver(itemJarResolvers))
-    } else {
-      null
+  private fun getResourceResolver(layoutComponent: ResolvedLayoutComponent): NamedResourceResolver? {
+    if (!layoutComponent.isClasspathable) {
+      return null
+    }
+    val itemJarResolvers = layoutComponent.resolveClasspaths().map {
+      NamedResourceResolver(
+        layoutComponent.name + "#" + it.relativePath, JarFilesResourceResolver(it.toList())
+      )
+    }
+    return NamedResourceResolver(layoutComponent.name, CompositeResourceResolver(itemJarResolvers))
+  }
+
+  private fun logUnavailableClasspath(failedComponents: List<ResolvedLayoutComponent>) {
+    val logMsg = failedComponents.joinToString("\n") {
+      val cp = it.getClasspaths().joinToString(", ")
+      "Layout component '${it.name}' has some nonexistent 'classPath' elements: '$cp'"
     }
+    LOG.atDebug().log(logMsg)
```

Comment:
why debug and not warn?

Reply:
It could still also affect verification results in theory, right? And whilte it's not actionable, it's important for user/us to understand what might be possibly wrong with the given IDE distribution (and JB does not control all of them).